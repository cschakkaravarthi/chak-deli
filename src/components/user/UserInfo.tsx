import React, { FC } from 'react';
import get from 'lodash.get';
import UserInfoModel, { Phone } from '../../shared/models/UserInfo.model';

type Props = {
  usersInfo?: UserInfoModel;
};

const UserInfo: FC<Props> = props => {
  const { usersInfo } = props;

  if (!usersInfo || !usersInfo.networkID) return null;

  return (
    <>
      <div className='user__info'>
        <div className='user__name'>
          <a href='#'>
            {usersInfo.lastName}, {usersInfo.firstName}
          </a>
        </div>
        <div className='user__title'>
          <p>{usersInfo.jobTitle}</p>
          <p>{usersInfo.department}</p>
        </div>
        <div className='user__email'>
          <a type='email' href={`mailto:${usersInfo.email}`}>
            {usersInfo.email}
          </a>
        </div>
        {get(usersInfo, 'location.office') && (
          <div className='user__office'>
            <p>Office: {get(usersInfo, 'location.office', '')}</p>
          </div>
        )}
        <div className='user__address'>
          <p>
            {get(usersInfo, 'location.room', '')} <br />
            {get(usersInfo, 'location.address', '')} <br />
            {`${get(usersInfo, 'location.city', '')}, ${get(usersInfo, 'location.state', '')} `}
            {get(usersInfo, 'location.postalCode', '')}
          </p>
        </div>
        <div className='user__phone'>
          {get(usersInfo, 'phone', [] as Phone[]).map((item: Phone, index: number) => (
            <p key={index}>
              {item.type === 'Desk' && <span>Phone: {item.num}</span>}
              {item.type === 'Mobile' && <span>Mobile: {item.num}</span>}
            </p>
          ))}
        </div>
      </div>
    </>
  );
};

export default UserInfo;
