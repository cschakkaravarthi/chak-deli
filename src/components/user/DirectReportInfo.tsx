import React, { FC } from 'react';

import { RouteComponentProps } from 'react-router-dom';
import { DirectReports } from '../../shared/models/UserInfo.model';
import images from '../../images/images';

interface Props {
  history?: any;
  directReports?: DirectReports[];
}

const DirectReportInfo: FC<Props> = props => {
  const { history, directReports } = props;

  if (!directReports || !directReports.length) return null;

  const handleDirectReportClick = (index: number): RouteComponentProps => {
    const { push } = history;

    return push(`/user/${directReports && directReports[index].networkID}`);
  };

  return (
    <div className='direct-report-info__container'>
      <div className='direct-report__icon'>
        <img src={images.caretLeft} width={20} height={20} />
      </div>
      <div className='direct-report__title'>
        <p>Direct Reports:</p>
      </div>
      <div className='direct-report__info'>
        <div className='direct-report__name'>
          {directReports &&
            directReports.map((item, index) => (
              <button
                className='button--anchor'
                key={`${item.networkID}${index}`}
                onClick={() => handleDirectReportClick(index)}
              >
                {item.firstName}, {item.lastName}
              </button>
            ))}
        </div>
      </div>
    </div>
  );
};

export default DirectReportInfo;
