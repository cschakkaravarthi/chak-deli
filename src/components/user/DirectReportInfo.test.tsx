import React from 'react';
import TestRenderer from 'react-test-renderer';
import DirectReportInfo from './DirectReportInfo';

import { dummyUserInfo } from '../../shared/models/UserInfo.model';

describe('DirectReportInfo', () => {
  it('renders correctly without info', () => {
    const tree = TestRenderer.create(<DirectReportInfo directReports={[]} />).toJSON();
    expect(tree).toBe(null);
  });

  it('renders correctly with info', () => {
    const tree = TestRenderer.create(<DirectReportInfo directReports={dummyUserInfo.directReports} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
