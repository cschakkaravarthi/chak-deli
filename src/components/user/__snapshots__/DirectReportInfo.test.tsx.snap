// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectReportInfo renders correctly with info 1`] = `
<div
  className="direct-report-info__container"
>
  <div
    className="direct-report__icon"
  >
    <img
      height={20}
      src="test-file-stub"
      width={20}
    />
  </div>
  <div
    className="direct-report__title"
  >
    <p>
      Direct Reports:
    </p>
  </div>
  <div
    className="direct-report__info"
  >
    <div
      className="direct-report__name"
    >
      <button
        className="button--anchor"
        onClick={[Function]}
      >
        Artrina
        , 
        Hines
      </button>
      <button
        className="button--anchor"
        onClick={[Function]}
      >
        Ann
        , 
        Vanderberg
      </button>
      <button
        className="button--anchor"
        onClick={[Function]}
      >
        Apollo
        , 
        Capito
      </button>
      <button
        className="button--anchor"
        onClick={[Function]}
      >
        Christopher
        , 
        Gillis
      </button>
    </div>
  </div>
</div>
`;
